bin/install_uninstall.ml

```diff
@@ -295,10 +301,20 @@ module File_ops_real (W : Workspace) : File_operations = struct
       | Error (e, _, _) ->
         User_message.prerr (User_error.make [ Pp.text (Unix.error_message e) ])
       | _ ->
-        User_error.raise
-          [ Pp.textf "Please delete non-empty directory %s manually."
-              (Path.to_string_maybe_quoted dir)
-          ]
+        let dir = Path.to_string_maybe_quoted dir in
+        if try_ then
+          User_message.prerr
+            (User_error.make
+               [ Pp.textf "Directory %s is not empty, cannot delete (ignoring)."
+                   dir
+               ])
+        else
+          User_error.raise
+            [ Pp.textf "Please delete non-empty directory %s manually." dir ]
+
+  let try_remove_dir_if_empty dir = remove_dir_if_empty ~try_:true dir
+
+  let remove_dir_if_empty dir = remove_dir_if_empty ~try_:false dir
```

Comment:
I wonder if these aliases that just partially apply `try_` are useful for readability. A flag like `on_non_empty:[Fail | Warn]` instead of `try_` and without the aliases would be shorter and just as readable.

Reply:
Good idea. I pushed a commit in this direction. Do you mind taking another look? Thanks!